### PR TITLE
Merge game-manager into master

### PR DESCRIPTION
**Pull request by strburst**
I think now is a good time to merge all the work I've done so far in implementing the game manager into master, before the two branches diverge too much. I've basically done a bunch of things:
- Make a bunch of enums for resources, terrain, etc., and move all the enums in `Config` into their own files. Also, put all the enums into their own package!
- Add a `Shop` class for managing the shop's stocks and prices.
- Add a `Tile` class for keeping information about terrain, whether a MULE is installed in a given spot, etc.
- Rename `Map` to `WorldMap`
- Miscellaneous changes

### CODE REVIEW

File: src/main/java/io/github/mountainrange/mule/Player.java [hunk 1]
```diff
@@ -1,25 +1,42 @@
...
+	private Race race;
+	private Color color = Color.BLACK;
+
 	private final int id;
 	private String name;
-	private Config.Race race;
-	private Color color = Color.BLACK;
 
-	public Player(int id, String name, Config.Race race, Color color) {
+	private int money;
```
Comment (jgkamat) [inline src/main/java/io/github/mountainrange/mule/Player.java:26]: Not sure about this, should we have money be a float or an int?

File: src/main/java/io/github/mountainrange/mule/WorldMap.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package io.github.mountainrange.mule;
+
+/**
+ * Holds information about the entire map.
+ */
+public class WorldMap {
+
+	Tile[][] tiles;
```
Comment (jgkamat) [inline src/main/java/io/github/mountainrange/mule/WorldMap.java:8]: Is this ever instantiated? (Its a little weird to have a class just for this)

File: src/main/java/io/github/mountainrange/mule/Player.java (hunk 1 above)
Comment (strburst) [inline src/main/java/io/github/mountainrange/mule/Player.java:26]: Money is always an integer in the original MULE game, so it should be fine. Eg prices in auctions only go up one at a time, and they don't have concepts like interest or anything.

File: src/main/java/io/github/mountainrange/mule/WorldMap.java (hunk 2 above)
Comment (strburst) [inline src/main/java/io/github/mountainrange/mule/WorldMap.java:8]: It's a skeleton right now; we should put all the information about terrain and who owns what in here (see also `Tile`). The map generator can use/instantiate it when we implement it. We should also read from this class when drawing things (e.g. the code in `Grid`).

Comment (strburst) [inline src/main/java/io/github/mountainrange/mule/WorldMap.java]: The map generator can use/instantiate it when we implement that.

Comment (jgkamat) [inline src/main/java/io/github/mountainrange/mule/WorldMap.java:8]: ah got it!

**Pull request merged**